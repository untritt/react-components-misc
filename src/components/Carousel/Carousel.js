import React from "react";
import cn from "classnames";

export default class Carousel extends React.Component {
  state = {
    activeItemIndex: 0
  };

  onClickPrev = e => {
    e.preventDefault();
    const { length } = this.props.paths;
    const prevIndex = (this.state.activeItemIndex + (length - 1)) % length;
    this.setState({
      activeItemIndex: prevIndex
    });
  };

  onClickNext = e => {
    e.preventDefault();
    const { length } = this.props.paths;
    const nextIndex = (this.state.activeItemIndex + 1) % length;
    this.setState({
      activeItemIndex: nextIndex
    });
  };

  renderImages() {
    return this.props.paths.map((image, index) => (
      <div
        className={cn({
          "carousel-item": true,
          active: this.state.activeItemIndex === index
        })}
        key={index}
      >
        <img alt="" className="d-block w-100" src={image} />
      </div>
    ));
  }

  render() {
    return (
      <div id="carousel" className="carousel slide" data-ride="carousel">
        <div className="carousel-inner">{this.renderImages()}</div>
        <a
          className="carousel-control-prev"
          href="#carousel"
          role="button"
          data-slide="prev"
          onClick={this.onClickPrev}
        >
          <span className="carousel-control-prev-icon" />
          <span className="sr-only">Previous</span>
        </a>
        <a
          className="carousel-control-next"
          href="#carousel"
          role="button"
          data-slide="next"
          onClick={this.onClickNext}
        >
          <span className="carousel-control-next-icon" />
          <span className="sr-only">Next</span>
        </a>
      </div>
    );
  }
}
