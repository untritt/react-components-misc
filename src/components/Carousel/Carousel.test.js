import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Carousel from "./Carousel";

Enzyme.configure({ adapter: new Adapter() });

test("Carousel", () => {
  const paths = [
    "/images/first.jpeg",
    "/images/second.jpeg",
    "/images/third.jpeg"
  ];
  const wrapper = mount(<Carousel paths={paths} />);
  expect(wrapper.render()).toMatchSnapshot();

  const next = wrapper.find(".carousel-control-next");
  const prev = wrapper.find(".carousel-control-prev");

  next.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  next.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  next.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  prev.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  prev.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  prev.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  prev.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();
});
