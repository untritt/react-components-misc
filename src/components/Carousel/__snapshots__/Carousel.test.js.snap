// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel 1`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 2`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 3`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 4`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 5`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 6`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 7`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`Carousel 8`] = `
<div
  class="carousel slide"
  data-ride="carousel"
  id="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/first.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/second.jpeg"
      >
        
      </img>
    </div>
    <div
      class="carousel-item active"
    >
      <img
        alt=""
        class="d-block w-100"
        src="/images/third.jpeg"
      >
        
      </img>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    data-slide="prev"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-prev-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    data-slide="next"
    href="#carousel"
    role="button"
  >
    <span
      class="carousel-control-next-icon"
    >
      
    </span>
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
